examples/async_printing.py

```diff
@@ -0,0 +1,203 @@
+            alerts.append(ALERTS[self._alert_count])
+            self._alert_count += 1
+            self._next_alert_time = cur_time + 4
+
+        else:
+            rand_num = random.randint(1, 20)
+            if rand_num > 2:
+                return []
+
+            for i in range(0, rand_num):
+                self._alert_count += 1
+                alerts.append("Alert {}".format(self._alert_count))
+
+            self._next_alert_time = 0
+
+        return alerts
+
+    def _generate_alert_str(self) -> str:
+        """
+        Combines alerts into one string that can be printed to the terminal
+        :return: the alert string
+        """
+        global ALERTS
+
+        alert_str = ''
+        alerts = self._get_alerts()
+
+        longest_alert = max(ALERTS, key=len)
+        num_astericks = len(longest_alert) + 8
+
+        for i, cur_alert in enumerate(alerts):
+            # Use padding to center the alert
+            padding = ' ' * int((num_astericks - len(cur_alert)) / 2)
+
+            if i > 0:
+                alert_str += '\n'
+            alert_str += '*' * num_astericks + '\n'
+            alert_str += padding + cur_alert + padding + '\n'
+            alert_str += '*' * num_astericks + '\n'
+
+        return alert_str
+
+    def _generate_colored_prompt(self) -> str:
+        """
+        Randomly generates a colored the prompt
+        :return: the new prompt
+        """
+        rand_num = random.randint(1, 20)
+
+        status_color = Fore.RESET
+
+        if rand_num == 1:
+            status_color = Fore.LIGHTRED_EX
+        elif rand_num == 2:
+            status_color = Fore.LIGHTYELLOW_EX
+        elif rand_num == 3:
+            status_color = Fore.CYAN
+        elif rand_num == 4:
+            status_color = Fore.LIGHTGREEN_EX
+        elif rand_num == 5:
+            status_color = Fore.LIGHTBLUE_EX
+
+        return status_color + self.visible_prompt + Fore.RESET
+
+    def _alerter_thread_func(self) -> None:
+        """ Prints alerts and updates the prompt any time the prompt is showing """
+
+        self._alert_count = 0
+        self._next_alert_time = 0
+
+        while not self._stop_thread:
+            # Always acquire terminal_lock before printing alerts or updating the prompt
+            # To keep the app responsive, do not block on this call
+            if self.terminal_lock.acquire(blocking=False):
+
+                # Get any alerts that need to be printed
+                alert_str = self._generate_alert_str()
+
+                # Generate a new prompt
+                new_prompt = self._generate_colored_prompt()
+
+                # Check if we have alerts to print
+                if alert_str:
+                    # new_prompt is an optional parameter to async_alert()
+                    self.async_alert(alert_str, new_prompt)
+                    new_title = "Alerts Printed: {}".format(self._alert_count)
+                    self.set_window_title(new_title)
```

Comment:
Thanks for adding usage of set_window_title to your example and info on it to the Sphinx docs